src/routing_node.rs

```diff
@@ -189,7 +189,8 @@ impl RoutingNode {
                     // the concept of a first node, as it is just hidden, not logically removed
                     // refactoring to crust 0.3 has made this logic even worse than it was.
                     if self.core.is_node() {
```

Comment:
can we get rid of this self.core.is_node() altogether; directly go to handle_on_connect() and inside that function we can send a (client_)hello (if we are a client)